### PR TITLE
Shivansh | Test | changed inventory sidebar design

### PR DESCRIPTION
**Pull request by shivanshRawat14**
https://app.clickup.com/t/29b9h5n

### CODE REVIEW

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.scss [hunk 1]
```diff
@@ -40,13 +53,81 @@
...
+                            filter: var(--color-image-filter-quaternary);
+                        }
+                    }
+
+                    .sidebar-list-icon {
+                        position: absolute;
+                        bottom: 9px;
+                        left: 220px;
+                    }
+                    .subOption {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.scss]:
```suggestion
                    .sub-option {
```

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts [hunk 2]
```diff
@@ -3,69 +3,182 @@ import { Component, EventEmitter, Output, OnDestroy } from '@angular/core';
...
+    icons: 'manufacturing.svg'
+  },
+  {
+    name: 'Warehouse Opening Balance',
+    icons: 'warehouse-opening-balance.svg'
+  },
+];
+
+/** Flat node with expandable and level information */
+interface ExampleFlatNode {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts]: rename this

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts [hunk 3]
```diff
@@ -3,69 +3,182 @@ import { Component, EventEmitter, Output, OnDestroy } from '@angular/core';
...
-    /**
-     * Releases the memory
-     *
-     * @memberof InventorySidebarComponent
-     */
-    public ngOnDestroy(): void {
-        this.destroyed$.next(true);
-        this.destroyed$.complete();
-    }
+  private _transformer = (node: SidebarNode, level: number) => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts]: don't use underscore

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts [hunk 4]
```diff
@@ -3,69 +3,182 @@ import { Component, EventEmitter, Output, OnDestroy } from '@angular/core';
...
+  }
+  
+  /**
+   * Function for routing list items by mat-tree-node
+   * 
+   * @param {string} listName holds the value of item clicked on sidebar menu
+   * @param {number} index is the index of the item
+   * @memberof InventorySidebarComponent
+   */
+  public gotoPage(listName: string, index: number): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts]:
```suggestion
  public gotoPage(node: any): void {
```

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts [hunk 5]
```diff
@@ -3,69 +3,182 @@ import { Component, EventEmitter, Output, OnDestroy } from '@angular/core';
...
+    // for Create New under Stock
+    if (listName == this.dataList[0].children[0].name && index == this.dataList[0].children[0].index) {
+      this.router.navigate(['/pages/new-inventory/stock/product/create'])
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts]: use node.link

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html [hunk 6]
```diff
@@ -431,6 +431,12 @@
                                                                                 class="icon-button-remove"
                                                                                 type="button"
                                                                                 (click)="deletePurchaseUnitPrice(i)"
+                                                                                *ngIf="
+                                                                                    i !==
+                                                                                    stockForm.purchaseAccountDetails
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html:435]: apply condition to delete only if records are more than 1

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html [hunk 7]
```diff
@@ -551,6 +559,12 @@
                                                                                 class="icon-button-remove"
                                                                                 type="button"
                                                                                 (click)="deleteSalesUnitPrice(i)"
+                                                                                *ngIf="
+                                                                                    i !==
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html]: apply condition to delete only if records are more than 1

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts [hunk 8]
```diff
@@ -160,6 +160,8 @@ export class StockCreateEditComponent implements OnInit, OnDestroy {
     public ngOnInit(): void {
         /* added image path */
         this.imgPath = isElectron ? 'assets/images/' : AppUrl + APP_FOLDER + 'assets/images/';
+        /** added parent class to body after entering new-inventory page */
+        document.querySelector("body").classList.add("new-inventory-create-stock");
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts]:
```suggestion
        document.querySelector("body").classList.add("stock-create-edit");
```

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts [hunk 9]
```diff
@@ -195,6 +197,8 @@ export class StockCreateEditComponent implements OnInit, OnDestroy {
     public ngOnDestroy(): void {
         this.destroyed$.next(true);
         this.destroyed$.complete();
+        /** remove parent class from body after exiting new-inventory page */
+        document.querySelector("body").classList.remove("new-inventory-create-stock");
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts]:
```suggestion
        document.querySelector("body").classList.remove("stock-create-edit");
```

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 10]
```diff
@@ -544,6 +544,17 @@ input {
             border: 1px solid var(--color-secondary-input-bar-border) !important;
         }
     }
+    .help-support-aside {
+        .inventory-sidebar {
+            .inventory-items {
+                .arrow-btn {
+                    .mat-ripple {
+                        display: none;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/css/style-2.scss]: use disableRipple true

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 11]
```diff
@@ -7736,6 +7747,80 @@ giddh-datepicker {
     }
 }
 
+.new-inventory-create-stock {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/css/style-2.scss]:
```suggestion
.stock-create-edit {
```

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 12]
```diff
@@ -7736,6 +7747,80 @@ giddh-datepicker {
...
+                                        text-field {
+                                            position: relative;
+                                            left: 7px;
+                                            input {
+                                                width: 112px;
+                                                height: 32px;
+                                            }
+                                        }
+                                        .icon-button-add {
+                                            .mat-ripple {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/css/style-2.scss]: use disableRipple

File: apps/web-giddh/src/assets/fonts/icomoon/icomoon.css [hunk 13]
```diff
@@ -26,855 +26,1171 @@
     -webkit-font-smoothing: antialiased;
     -moz-osx-font-smoothing: grayscale;
 }
-.icon-attachment:before {
-    content: "\e900";
-    color: #999da0;
-}
-.icon-download-new:before {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/fonts/icomoon/icomoon.css:33]: check this

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.scss (hunk 1 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.scss]: changed

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts (hunk 2 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts]: renamed

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts (hunk 3 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts]: underscore removed

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts (hunk 4 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts (hunk 5 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/new-inventory/component/inventory-sidebar/inventory-sidebar.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html (hunk 6 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html:435]: done

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html (hunk 7 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html]: done

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts (hunk 8 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts (hunk 9 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts]: done

File: apps/web-giddh/src/assets/css/style-2.scss (hunk 11 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/assets/css/style-2.scss]: done

File: apps/web-giddh/src/assets/css/style-2.scss (hunk 10 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/assets/css/style-2.scss]: used

File: apps/web-giddh/src/assets/css/style-2.scss (hunk 12 above)
Comment (shivanshRawat14) [reply apps/web-giddh/src/assets/css/style-2.scss]: used

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**